### PR TITLE
Show about dialog

### PR DESCRIPTION
**Pull request by metsma**
WE2-465

Signed-off-by: Raul Metsma <raul@metsma.ee>

### CODE REVIEW

File: src/mac/shared.hpp [hunk 1]
```diff
@@ -22,10 +22,35 @@
...
 #if __MAC_OS_X_VERSION_MIN_REQUIRED < 110000
 NSString* const SFExtensionMessageKey = @"message";
 #endif
+
+static NSUserDefaults* getUserDefaults()
+{
+    return [[NSUserDefaults alloc] initWithSuiteName:WebEidShared];
+}
+
+static void setValue(NSString* key, id value)
```
Comment (mrts) [inline src/mac/shared.hpp:39]: Glad to see DRY code, cheers!

File: src/mac/safari-extension.mm [hunk 2]
```diff
@@ -71,13 +68,18 @@ - (void)beginRequestWithExtensionContext:(NSExtensionContext *)context
     NSLog(@"web-eid-safari-extension: web-eid-safari isRunning: %d", isRunning);
     if (!isRunning) {
         if (NSURL *appURL = [NSWorkspace.sharedWorkspace URLForApplicationWithBundleIdentifier:WebEidApp]) {
+            setValue(WebEidStarting, @(true));
             if ([NSWorkspace.sharedWorkspace launchApplication:appURL.path]) {
                 NSLog(@"web-eid-safari-extension: started app");
+                do {
+                    [NSThread sleepForTimeInterval:1.0f];
+                    NSLog(@"web-eid-safari-extension: waiting to be running %@", [getUserDefaults() objectForKey:WebEidStarting]);
+                } while ([getUserDefaults() boolForKey:WebEidStarting]);
```
Comment (mrts) [inline src/mac/safari-extension.mm:77]: Great, so it worked out as we discussed! However, we need to add a timeout here, otherwise this may result in an infinite loop in case of errors in `main()` above so that `takeValue(WebEidStarting)` is either not reached or fails.

File: src/ui/webeiddialog.cpp [hunk 3]
```diff
@@ -355,6 +367,7 @@ bool WebEidDialog::event(QEvent* event)
 {
     if (event->type() == QEvent::LanguageChange) {
         ui->retranslateUi(this);
+        ui->versionLabel->setText(tr("Version: %1").arg(qApp->applicationVersion()));
```
Comment (mrts) [inline src/ui/webeiddialog.cpp:370]: Why is this needed here in addition to the same line in constructor?

**Review by mrts — CHANGES_REQUESTED** [review]
Great work as always, thanks! Timeout needs to be added to the while loop that checks `WebEidStarting`, but this is no big deal.

File: src/ui/webeiddialog.cpp (hunk 3 above)
Comment (metsma) [reply src/ui/webeiddialog.cpp:370]: retranslation might happen later or not at all

File: src/mac/safari-extension.mm (hunk 2 above)
Comment (metsma) [reply src/mac/safari-extension.mm]: Added timout

**Review by mrts — APPROVED** [review]
